diff -ruN NVIDIA-Linux-x86_64-304.137p1/kernel/nv.c NVIDIA-Linux-x86_64-304.137p2/kernel/nv.c
--- NVIDIA-Linux-x86_64-304.137p1/kernel/nv.c	2022-01-10 19:56:50.706995715 -0300
+++ NVIDIA-Linux-x86_64-304.137p2/kernel/nv.c	2022-01-10 20:07:39.959220633 -0300
@@ -3327,8 +3327,12 @@
 
 #if defined(CONFIG_VGA_ARB)
 #if defined(VGA_DEFAULT_DEVICE)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
+    vga_get(VGA_DEFAULT_DEVICE, VGA_RSRC_LEGACY_MASK, 0);
+#else
     vga_tryget(VGA_DEFAULT_DEVICE, VGA_RSRC_LEGACY_MASK);
 #endif
+#endif
     vga_set_legacy_decoding(dev, VGA_RSRC_NONE);
 #endif
 
diff -ruN NVIDIA-Linux-x86_64-304.137p1/kernel/nv-linux.h NVIDIA-Linux-x86_64-304.137p2/kernel/nv-linux.h
--- NVIDIA-Linux-x86_64-304.137p1/kernel/nv-linux.h	2022-01-10 19:57:07.377477338 -0300
+++ NVIDIA-Linux-x86_64-304.137p2/kernel/nv-linux.h	2022-01-10 20:09:33.516806031 -0300
@@ -152,8 +152,10 @@
 
 #if defined(NVCPU_X86_64) && !defined(KERNEL_2_4) && !defined(HAVE_COMPAT_IOCTL)
 #include <linux/syscalls.h>         /* sys_ioctl()                      */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 9, 0)
 #include <linux/ioctl32.h>          /* register_ioctl32_conversion()    */
 #endif
+#endif
 
 #if defined(NVCPU_X86_64) && defined(KERNEL_2_4)
 #include <asm/ioctl32.h>            /* sys_ioctl() (ioctl32)            */
@@ -1979,10 +1981,13 @@
                                             pages, vmas, NULL);
 
         #else
-
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
+               return get_user_pages_remote(mm, start, nr_pages, flags,
+                                            pages, vmas, NULL);
+	#else
                return get_user_pages_remote(tsk, mm, start, nr_pages, flags,
                                             pages, vmas);
-
+	#endif
         #endif
 
         }
