diff -ruN NVIDIA-Linux-x86_64-304.137p1/kernel/os-interface.c NVIDIA-Linux-x86_64-304.137p2/kernel/os-interface.c
--- NVIDIA-Linux-x86_64-304.137p1/kernel/os-interface.c	2022-01-10 19:56:53.451468695 -0300
+++ NVIDIA-Linux-x86_64-304.137p2/kernel/os-interface.c	2022-01-10 20:34:22.677245200 -0300
@@ -772,7 +772,11 @@
         // the requested timeout has expired, loop until less
         // than a jiffie of the desired delay remains.
         //
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 14, 0)
+        current->__state = TASK_INTERRUPTIBLE;
+#else
         current->state = TASK_INTERRUPTIBLE;
+#endif
         do
         {
             schedule_timeout(jiffies);
