--- 1/kernel/nv-linux.h	2017-09-14 17:51:09.000000000 -0300
+++ 2/kernel/nv-linux.h	2021-12-25 20:26:19.821461438 -0300
@@ -134,7 +134,9 @@
 #include <asm/tlbflush.h>           /* flush_tlb(), flush_tlb_all()     */
 #include <linux/cpu.h>              /* CPU hotplug support              */
 #endif
-#include <asm/kmap_types.h>         /* page table entry lookup          */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 11, 0)
+	#include <asm/kmap_types.h>         /* page table entry lookup          */
+#endif
 #endif
 
 #include <linux/pci.h>              /* pci_find_class, etc              */
