--- NVIDIA-Linux-x86_64-304.137mod2/kernel/conftest.sh	2022-01-08 17:42:05.315884362 -0300
+++ NVIDIA-Linux-x86_64-304.137mod1/kernel/conftest.sh	2022-01-08 17:52:56.140500898 -0300
@@ -342,6 +342,8 @@
             # Determine if the set_memory_uc() function is present.
             #
             CODE="
+            #include <asm/page.h>
+            #include <asm/pgtable.h>
             #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
             #include <asm/set_memory.h>
             #else
@@ -359,6 +361,8 @@
             # Determine if the set_memory_array_uc() function is present.
             #
             CODE="
+            #include <asm/page.h>
+            #include <asm/pgtable.h>
             #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
             #include <asm/set_memory.h>
             #else
@@ -376,6 +380,8 @@
             # Determine if the set_pages_uc() function is present.
             #
             CODE="
+            #include <asm/page.h>
+            #include <asm/pgtable.h>
             #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
             #include <asm/set_memory.h>
             #else
